subprojects/shipkit/src/test/groovy/org/shipkit/internal/util/GitHubStatusCheckTest.groovy

```diff
@@ -0,0 +1,38 @@
+package org.shipkit.internal.util
+
+import org.shipkit.internal.gradle.versionupgrade.MergePullRequestTask
+import spock.lang.Specification
+
+class GitHubStatusCheckTest extends Specification {
+
+    MergePullRequestTask task = Mock(MergePullRequestTask)
+    GitHubApi gitHubApi = Mock(GitHubApi)
+
+    GitHubStatusCheck gitHubStatusCheck = new GitHubStatusCheck(task, gitHubApi)
+
+    def "should return true if status true before timeout"() {
+        given:
+        task.getPullRequestSha() >> "sha"
+        task.getUpstreamRepositoryName() >> "upstreamRepo"
+        1 * gitHubApi.get("/repos/upstreamRepo/commits/sha/status") >> "{\"state\": \"pending\"}"
+        1 * gitHubApi.get("/repos/upstreamRepo/commits/sha/status") >> "{\"state\": \"success\"}"
+        when:
+        def result = gitHubStatusCheck.checkStatusWithTimeout()
+        then:
+        result == true
+    }
+
+    def "should return true if has error status"() {
```

Comment:
Test name lies, it actually throws an exception. Also there is no test for when checkStatusWithTimeout returns false.

Reply:
Due to retries, such a test would be very long and don't bring much value. I'll change it to set up amount of retries in constructor.

Reply:
Maybe it would be good to put Thread.sleep into a separate class, and mock it?